### PR TITLE
Resolve build hang when docker daemon under load

### PR DESCRIPTION
**Pull request by jim-minter**
Resolves https://bugzilla.redhat.com/show_bug.cgi?id=1442875
[test][testextended][extended:core(builds)]

### CODE REVIEW

**Comment by jim-minter** [comment]
There were other bugs, but I think flake #13694 was seen at https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended/236/consoleFull#110463848456cbb9a5e4b02b88ae8c2f77

**Comment by jim-minter** [comment]
@bparees ptal
@openshift/devex

File: pkg/build/builder/sti.go [hunk 1]
```diff
@@ -40,7 +41,11 @@ type runtimeBuilderFactory struct{}
 
 // Builder delegates execution to S2I-specific code
 func (_ runtimeBuilderFactory) Builder(config *s2iapi.Config, overrides s2ibuild.Overrides) (s2ibuild.Builder, s2iapi.BuildInfo, error) {
-	builder, buildInfo, err := s2i.Strategy(config, overrides)
+	client, err := docker.NewEngineAPIClient(config.DockerConfig)
```
Comment (gabemontero) [inline pkg/build/builder/sti.go]: A comment somehow connecting the dots to the docker daemon load issue might be useful months from now when someone looks at this and asks why we don't construct the client just once.

File: pkg/build/builder/sti.go [hunk 2]
```diff
@@ -259,7 +264,11 @@ func (s *S2IBuilder) Build() error {
 		return errors.New(buffer.String())
 	}
 
-	glog.V(4).Infof("Creating a new S2I builder with build config: %#v\n", describe.Config(config))
+	client, err := docker.NewEngineAPIClient(config.DockerConfig)
```
Comment (gabemontero) [inline pkg/build/builder/sti.go]: A comment somehow connecting the dots to the docker daemon load issue might be useful months from now when someone looks at this and asks why we don't construct the client just once.

Comment (bparees) [inline pkg/build/builder/sti.go]: +1

**Comment by bparees** [comment]
agree w/ @gabemontero's suggestion, otherwise lgtm.

File: pkg/build/builder/sti.go (hunk 2 above)
Comment (jim-minter) [inline pkg/build/builder/sti.go]: Hmm, the docker daemon load issue is dealt with in the bowels of s2i; this PR is really just the bump.  The API changed to require passing in a docker client, hence this diff.  I'm not sure that really merits a comment (he says, having just written this explanation here).

Comment (bparees) [inline pkg/build/builder/sti.go]: sorry the git diff tricked me into thinking both of these changes were in the same function and thus someone would expect to reuse the same client object.

**Comment by bparees** [comment]
[merge]

**Pull request merged**